File: server/src/routes/logRoutes.js
/**
 * 日志相关路由
 * @module routes/logRoutes
 * @desc 定义日志相关的API端点
 */

const express = require('express');
const router = express.Router();
const logController = require('../controllers/logController');

/**
 * 接收前端发送的操作日志
 * @name POST /api/logs
 * @function
 * @memberof module:routes/logRoutes
 * @inner
 * @param {Object} req.body - 日志数据
 */
router.post('/', logController.receiveLog);

/**
 * 获取日志列表（管理员功能）
 * @name GET /api/logs
 * @function
 * @memberof module:routes/logRoutes
 * @inner
 * @param {Object} req.query - 查询参数
 * @param {number} req.query.limit - 限制返回数量
 * @param {number} req.query.offset - 偏移量
 * @param {string} req.query.type - 日志类型
 * @param {string} req.query.startDate - 开始日期
 * @param {string} req.query.endDate - 结束日期
 * @param {string} req.query.username - 用户名
 */
router.get('/', logController.getLogsList);

/**
 * 获取日志统计信息（管理员功能）
 * @name GET /api/logs/stats
 * @function
 * @memberof module:routes/logRoutes
 * @inner
 * @param {Object} req.query - 查询参数
 * @param {string} req.query.startDate - 开始日期
 * @param {string} req.query.endDate - 结束日期
 */
router.get('/stats', logController.getLogStats);

/**
 * 清理过期日志（管理员功能）
 * @name DELETE /api/logs/clean
 * @function
 * @memberof module:routes/logRoutes
 * @inner
 * @param {Object} req.query - 查询参数
 * @param {number} req.query.days - 保留天数，默认90天
 */
router.delete('/clean', logController.cleanLogs);

module.exports = router;

File: server-go/internal/routes/log_routes.go
package routes

import (
	"homemoney/internal/handler"
	"homemoney/internal/service"

	"github.com/gin-gonic/gin"
)

// SetupLogRoutes 设置日志相关路由
// @Summary 设置日志路由
// @Description 配置日志相关的API端点
// @Tags 路由配置
func SetupLogRoutes(router *gin.RouterGroup, logService *service.LogService) {
	// 创建日志处理器
	logHandler := handler.NewLogHandler(logService)
	
	// 日志相关路由组
	logs := router.Group("/logs")
	{
		// 接收操作日志
		logs.POST("", logHandler.ReceiveLog)
		
		// 获取日志列表
		logs.GET("", logHandler.GetLogsList)
		
		// 获取日志统计信息
		logs.GET("/stats", logHandler.GetLogStats)
		
		// 清理过期日志
		logs.DELETE("/clean", logHandler.CleanLogs)
	}
}
